Connection error to standard address space in openshift 4.3

**Describe the bug**
When running enmasse on openshift 4.3 we have several errors of amqp clients connecting to standard address spaces, the most common ones are:

A vertx client in systemtests code get errors like this:
```
java.util.concurrent.ExecutionException: io.vertx.core.VertxException: Disconnected
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2022)
	at io.enmasse.systemtest.isolated.standard.SmokeTest.testAnycast(SmokeTest.java:219)
	at io.enmasse.systemtest.isolated.standard.SmokeTest.smoketest(SmokeTest.java:142)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:675)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:125)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:132)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:124)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:74)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:104)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:62)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:43)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:35)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:202)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:198)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:69)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:229)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:197)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:191)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:150)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: io.vertx.core.VertxException: Disconnected
	at io.vertx.proton.impl.ProtonClientImpl.lambda$null$0(ProtonClientImpl.java:81)
	at io.vertx.proton.impl.ProtonConnectionImpl.fireDisconnect(ProtonConnectionImpl.java:381)
	at io.vertx.proton.impl.ProtonTransport.handleSocketEnd(ProtonTransport.java:94)
	at io.vertx.core.net.impl.NetSocketImpl.lambda$new$2(NetSocketImpl.java:96)
	at io.vertx.core.streams.impl.InboundBuffer.handleEvent(InboundBuffer.java:235)
	at io.vertx.core.streams.impl.InboundBuffer.write(InboundBuffer.java:125)
	at io.vertx.core.net.impl.NetSocketImpl.handleClosed(NetSocketImpl.java:356)
	at io.vertx.core.net.impl.VertxHandler.lambda$channelInactive$4(VertxHandler.java:162)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:320)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:43)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:188)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:180)
	at io.vertx.core.net.impl.VertxHandler.channelInactive(VertxHandler.java:162)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:242)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:228)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:221)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelInactive(ChannelInboundHandlerAdapter.java:75)
	at io.netty.handler.timeout.IdleStateHandler.channelInactive(IdleStateHandler.java:277)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:242)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:228)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:221)
	at io.netty.handler.stream.ChunkedWriteHandler.channelInactive(ChunkedWriteHandler.java:141)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:242)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:228)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:221)
	at io.netty.handler.codec.ByteToMessageDecoder.channelInputClosed(ByteToMessageDecoder.java:390)
	at io.netty.handler.codec.ByteToMessageDecoder.channelInactive(ByteToMessageDecoder.java:355)
	at io.netty.handler.ssl.SslHandler.channelInactive(SslHandler.java:1076)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:242)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:228)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:221)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1403)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:242)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:228)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:912)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:827)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:495)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:905)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Other example, with cli-rhea we get this error:

```
cli-rhea-sender --broker test:test@XXXXXXX:443 --timeout 30 --log-lib TRANSPORT_FRM --address message-basiccli_rhea --msg-content msg no. %d --log-msgs json --conn-ssl true --count 10
2019-11-11T15:44:10.393Z INFO  [AbstractClient] io.enmasse.systemtest.messagingclients.rhea.RheaClientSender CLI_RHEA_SENDER Return code - 1
2019-11-11T15:44:10.393Z INFO  [AbstractClient] io.enmasse.systemtest.messagingclients.rhea.RheaClientSender CLI_RHEA_SENDER stdout : 
2019-11-11T15:44:10.393Z ERROR [AbstractClient] io.enmasse.systemtest.messagingclients.rhea.RheaClientSender CLI_RHEA_SENDER stderr : I1111 10:44:09.509270   12392 logs.go:49] (0xc4213ac0b0) (0xc4213ee140) Create stream
I1111 10:44:09.509372   12392 logs.go:49] (0xc4213ac0b0) (0xc4213ee140) Stream added, broadcasting: 1
I1111 10:44:09.540403   12392 logs.go:49] (0xc4213ac0b0) Reply frame received for 1
I1111 10:44:09.540424   12392 logs.go:49] (0xc4213ac0b0) (0xc4210340a0) Create stream
I1111 10:44:09.540430   12392 logs.go:49] (0xc4213ac0b0) (0xc4210340a0) Stream added, broadcasting: 3
I1111 10:44:09.570288   12392 logs.go:49] (0xc4213ac0b0) Reply frame received for 3
I1111 10:44:09.570328   12392 logs.go:49] (0xc4213ac0b0) (0xc42114c000) Create stream
I1111 10:44:09.570342   12392 logs.go:49] (0xc4213ac0b0) (0xc42114c000) Stream added, broadcasting: 5
I1111 10:44:09.603232   12392 logs.go:49] (0xc4213ac0b0) Reply frame received for 5
I1111 10:44:10.359953   12392 logs.go:49] (0xc4213ac0b0) Data frame received for 5
I1111 10:44:10.359970   12392 logs.go:49] (0xc42114c000) (5) Data frame handling
I1111 10:44:10.359976   12392 logs.go:49] (0xc42114c000) (5) Data frame sent
2019-11-11T15:44:10.740Z rhea:frames [connection-1] -> { protocol_id: 3, major: 1, minor: 0, revision: 0 }
I1111 10:44:10.366238   12392 logs.go:49] (0xc4213ac0b0) Data frame received for 5
I1111 10:44:10.366250   12392 logs.go:49] (0xc42114c000) (5) Data frame handling
I1111 10:44:10.366263   12392 logs.go:49] (0xc42114c000) (5) Data frame sent
2019-11-11T15:44:10.749Z rhea:frames [connection-1] <- { protocol_id: 3, major: 1, minor: 0, revision: 0 }
I1111 10:44:10.371892   12392 logs.go:49] (0xc4213ac0b0) Data frame received for 5
I1111 10:44:10.371904   12392 logs.go:49] (0xc42114c000) (5) Data frame handling
I1111 10:44:10.371910   12392 logs.go:49] (0xc42114c000) (5) Data frame sent
2019-11-11T15:44:10.755Z rhea:frames [connection-1]:0 <- sasl_mechanisms#40 {"sasl_server_mechanisms":[]} 
I1111 10:44:10.380330   12392 logs.go:49] (0xc4213ac0b0) Data frame received for 5
I1111 10:44:10.380349   12392 logs.go:49] (0xc42114c000) (5) Data frame handling
I1111 10:44:10.380354   12392 logs.go:49] (0xc42114c000) (5) Data frame sent
events.js:165
      throw err;
      ^

Error: Uncaught, unspecified "error" event. ([object Object])
    at Container.emit (events.js:163:17)
    at Container.dispatch (/usr/lib/node_modules/cli-rhea/node_modules/rhea/lib/container.js:42:33)
    at Connection.dispatch (/usr/lib/node_modules/cli-rhea/node_modules/rhea/lib/connection.js:235:40)
    at Connection.input (/usr/lib/node_modules/cli-rhea/node_modules/rhea/lib/connection.js:508:18)
    at emitOne (events.js:96:13)
    at TLSSocket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at TLSSocket.Readable.push (_stream_readable.js:134:10)
    at TLSWrap.onread (net.js:559:20)
I1111 10:44:10.389109   12392 logs.go:49] (0xc4213ac0b0) (0xc4210340a0) Stream removed, broadcasting: 3
I1111 10:44:10.389131   12392 logs.go:49] (0xc4213ac0b0) Data frame received for 1
I1111 10:44:10.389135   12392 logs.go:49] (0xc4213ee140) (1) Data frame handling
I1111 10:44:10.389140   12392 logs.go:49] (0xc4213ee140) (1) Data frame sent
I1111 10:44:10.389144   12392 logs.go:49] (0xc4213ac0b0) (0xc4213ee140) Stream removed, broadcasting: 1
I1111 10:44:10.389217   12392 logs.go:49] (0xc4213ac0b0) (0xc42114c000) Stream removed, broadcasting: 5
I1111 10:44:10.389243   12392 logs.go:49] (0xc4213ac0b0) Go away received
I1111 10:44:10.389276   12392 logs.go:49] (0xc4213ac0b0) (0xc4213ee140) Stream removed, broadcasting: 1
I1111 10:44:10.389283   12392 logs.go:49] (0xc4213ac0b0) (0xc4210340a0) Stream removed, broadcasting: 3
I1111 10:44:10.389287   12392 logs.go:49] (0xc4213ac0b0) (0xc42114c000) Stream removed, broadcasting: 5
command terminated with exit code 1
```

This errors have in common the same log error showed in qpid router logs:
```
2019-11-11 14:56:05.530523 +0000 AUTHSERVICE (warning) Downstream disconnected: amqp:connection:framing-error SSL Failure: error:14090086:SSL routines:ssl3_get_server_certificate:certificate verify failed
2019-11-11 14:56:05.530738 +0000 SERVER (info) [C53] Connection from 10.131.0.4:60812 (to 0.0.0.0:5671) failed: amqp:unauthorized-access Failed to authenticate client [mech=none]
```

**Additional context**
The complete logs can be seen in our CI ask me for the logs.

--- a/keycloak-plugin/src/main/sh/init-keycloak.sh
+++ b/keycloak-plugin/src/main/sh/init-keycloak.sh
@@ -2,6 +2,7 @@
 set -e
 KEYCLOAK_CONFIG=${KEYCLOAK_DIR}/standalone/configuration/
 OPENSHIFT_CA=${OPENSHIFT_CA:-/var/run/secrets/kubernetes.io/serviceaccount/ca.crt}
+SERVICE_CA=${OPENSHIFT_CA:-/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt}
 
 cp ${KEYCLOAK_PLUGIN_DIR}/configuration/* ${KEYCLOAK_CONFIG}/
 cp ${KEYCLOAK_PLUGIN_DIR}/configuration/${KEYCLOAK_CONFIG_FILE} ${KEYCLOAK_CONFIG}/standalone-openshift.xml
@@ -21,4 +22,6 @@ cp /etc/pki/java/cacerts ${TRUSTSTORE_PATH}
 chmod 644 ${TRUSTSTORE_PATH}
 echo "Copied system trust store. Importing OpenShift CA"
 keytool -import -noprompt -file ${OPENSHIFT_CA} -alias firstCA -deststorepass changeit -keystore $TRUSTSTORE_PATH
+echo "Importing OpenShift Service CA"
+keytool -import -noprompt -file ${SERVICE_CA} -alias secondCA -deststorepass changeit -keystore $TRUSTSTORE_PATH
 echo "Truststore ${TRUSTSTORE_PATH} created"

--- a/keycloak-user-api/src/main/java/io/enmasse/user/keycloak/KubeKeycloakFactory.java
+++ b/keycloak-user-api/src/main/java/io/enmasse/user/keycloak/KubeKeycloakFactory.java
@@ -22,11 +22,13 @@ import org.slf4j.LoggerFactory;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.MultivaluedMap;
 import java.io.ByteArrayInputStream;
+import java.io.File;
 import java.io.IOException;
 import java.io.OutputStream;
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Type;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 import java.security.KeyStore;
 import java.security.cert.CertificateFactory;
 import java.util.Base64;
@@ -39,6 +41,7 @@ public class KubeKeycloakFactory implements KeycloakFactory {
 
     private final ExecutorService executorService = Executors.newSingleThreadExecutor();
     private final NamespacedKubernetesClient kubeClient;
+    private static final File serviceCaFile = new File("/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt");
 
     public KubeKeycloakFactory(NamespacedKubernetesClient kubeClient) {
         this.kubeClient = kubeClient;
@@ -70,7 +73,13 @@ public class KubeKeycloakFactory implements KeycloakFactory {
         }
         Secret certificate = kubeClient.secrets().inNamespace(certificateSecretNamespace).withName(standard.getCertificateSecret().getName()).get();
 
-        KeyStore trustStore = createKeyStore(b64dec.decode(certificate.getData().get("tls.crt")));
+        byte []serviceCa = null;
+        try {
+            serviceCa = Files.readAllBytes(serviceCaFile.toPath());
+        } catch (IOException e) {
+            log.warn("Unable to load service CA, ignoring", e);
+        }
+        KeyStore trustStore = createKeyStore(b64dec.decode(certificate.getData().get("tls.crt")), serviceCa);
         ResteasyJackson2Provider provider = new ResteasyJackson2Provider() {
             @Override
             public void writeTo(Object value, Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType, MultivaluedMap<String, Object> httpHeaders, OutputStream entityStream) throws IOException {
@@ -97,14 +106,19 @@ public class KubeKeycloakFactory implements KeycloakFactory {
                 .build();
     }
 
-    private static KeyStore createKeyStore(byte [] ca) {
+    private static KeyStore createKeyStore(byte [] ca, byte [] serviceCa) {
         try {
             KeyStore keyStore = KeyStore.getInstance("JKS");
             keyStore.load(null);
             CertificateFactory cf = CertificateFactory.getInstance("X.509");
-            keyStore.setCertificateEntry("keycloak",
-                    cf.generateCertificate(new ByteArrayInputStream(ca)));
-
+            if (ca != null) {
+                keyStore.setCertificateEntry("keycloak",
+                        cf.generateCertificate(new ByteArrayInputStream(ca)));
+            }
+            if (serviceCa != null) {
+                keyStore.setCertificateEntry("service-ca",
+                        cf.generateCertificate(new ByteArrayInputStream(serviceCa)));
+            }
             return keyStore;
         } catch (Exception ignored) {
             log.warn("Error creating keystore. Ignoring", ignored);

